src/current/v23.2/live-migration-service.md

```diff
@@ -0,0 +1,612 @@
+
+For example:
+
+~~~ yaml
+lms:
+  sslVolumes:
+    - name: lms-ca
+      secret:
+        secretName: lms-ca
+    - name: lms-cert
+      secret:
+        secretName: lms-cert
+    - name: lms-key
+      secret:
+        secretName: lms-key
+  sslVolumeMounts:
+    - mountPath: "/app/certs"
+      name: lms-ca
+      readOnly: true
+    - mountPath: "/app/certs"
+      name: lms-cert
+      readOnly: true
+    - mountPath: "/app/certs"
+      name: lms-key
+      readOnly: true
+~~~
+
+Set the [corresponding config keys](#cert-var) to the same mount path. For example, within `config.json` in the [external secret](#manage-external-secret):
+
+~~~ yaml
+config.json: |
+  {
+    "SSL_CA": "/app/certs",
+    "SSL_CERT": "/app/certs",
+    "SSL_KEY": "/app/certs",
+  }
+~~~
+
+#### Configure source and target certificates
+
+Mount the source and target certificate, key, and (optional) CA certificate to separate volumes. Cockroach Labs recommends mounting certificates to `/app/certs`.
+
+For example:
+
+~~~ yaml
+lms:
+  sslVolumes:
+    - name: source-ca
+      secret:
+        secretName: source-ca
+    - name: source-cert
+      secret:
+        secretName: source-cert
+    - name: source-key
+      secret:
+        secretName: source-key
+    - name: cockroach-ca
+      secret:
+        secretName: cockroach-ca
+    - name: cockroach-cert
+      secret:
+        secretName: cockroach-cert
+    - name: cockroach-key
+      secret:
+        secretName: cockroach-key
+  sslVolumeMounts:
+    - mountPath: "/app/certs"
+      name: source-ca
+      readOnly: true
+    - mountPath: "/app/certs"
+      name: source-cert
+      readOnly: true
+    - mountPath: "/app/certs"
+      name: source-key
+      readOnly: true
+    - mountPath: "/app/certs"
+      name: cockroach-ca
+      readOnly: true
+    - mountPath: "/app/certs"
+      name: cockroach-cert
+      readOnly: true
+    - mountPath: "/app/certs"
+      name: cockroach-key
+      readOnly: true
+~~~
+
+Specify the corresponding client certificates and keys in the [database connection strings](#source-and-target-connections). For example, within `config.json` in the [external secret](#manage-external-secret):
+
+~~~ json
+config.json: |
+  {
+    "INIT_SOURCE": "mysql://{username}:{password}@{protocol}({host}:{port})/{database}?sslmode=verify-full?sslrootcert=path/to/mysql.ca&sslcert=path/to/mysql.crt&sslkey=path/to/mysql.key",
+    "INIT_TARGET": "postgresql://{username}:{password}@{host}:{port}/{database}?sslmode=verify-full?sslrootcert=path/to/ca.crt&sslcert=path/to/client.username.crt&sslkey=path/to/client.username.key"
+  }
+~~~
+
+#### Configure orchestrator and client certificates
+
+Mount the orchestrator certificate, key, and (optional) CA certificate to separate volumes. Cockroach Labs recommends mounting certificates to `/app/certs`.
+
+For example:
+
+~~~ yaml
```

Comment:
Actually for the TLS setup for orchestrator, it will actually look something like this:
```
sslVolumes:
     - name: orch-tls
       secret:
         secretName: orch-tls
  sslVolumeMounts:
    - mountPath: "/app/certs"
      name: orch-tls
      readOnly: true

  env:
    - name: ORCH_CA_TLS_CERT
      value: /app/certs/ca.crt
    - name: ORCH_TLS_CERT
      value: /app/certs/tls.crt
    - name: ORCH_TLS_KEY
      value: /app/certs/tls.key
```
Follow this diff again: https://github.com/cockroachdb/molt/compare/main...user/rluu/run-lms-secrets-and-tls

This does depend on having a separate `orch-tls` external secret that looks like this:
```

apiVersion: external-secrets.io/v1beta1
kind: ExternalSecret
metadata:
  name: orch-tls
spec:
  refreshInterval: 1h
  secretStoreRef:
    name: aws-secret-store
    kind: SecretStore
  target:
    name: orch-tls
    creationPolicy: Owner
    template:
      engineVersion: v2
      data:
        ca.crt: '{{ .caCert }}'
        tls.crt: '{{ .serverCert }}'
        tls.key: '{{ .serverKey }}'
  data:
  - secretKey: caCert
    remoteRef:
      key: orch-certs
      property: caCert
  - secretKey: serverCert
    remoteRef:
      key: orch-certs
      property: serverCert
  - secretKey: serverKey
    remoteRef:
      key: orch-certs
      property: serverKey
```

Reply:
As you can see in the example, you can do this through ENV variables as well. But the config .json you have below looks reasonable too.